src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -142,38 +143,40 @@ private string Encode(IClientAssertionCertificate credential)
         internal class JWTHeader
         {
             protected IClientAssertionCertificate Credential { get; private set; }
+            private string _type;
+            private string _alg;
 
             public JWTHeader(IClientAssertionCertificate credential)
             {
                 this.Credential = credential;
+                _alg = (this.Credential == null)
+                    ? JsonWebTokenConstants.Algorithms.None
+                    : JsonWebTokenConstants.Algorithms.RsaSha256;
+
+                _type = JsonWebTokenConstants.HeaderType;
             }
 
             [DataMember(Name = JsonWebTokenConstants.ReservedHeaderParameters.Type)]
-            public static string Type
+            public string Type
             {
                 get
                 {
-                    return JsonWebTokenConstants.HeaderType;
+                    return _type;
                 }
 
-                set
-                {
-                    // This setter is required by DataContractJsonSerializer
-                }
+                set { _type = value; }
             }
 
             [DataMember(Name = JsonWebTokenConstants.ReservedHeaderParameters.Algorithm)]
             public string Algorithm
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this an auto-implemented property and remove its backing field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS2292)